tests/appsec/iast/test_grpc_iast.py

```diff
@@ -148,3 +150,19 @@ class MyUserDict(UserDict):
             mutable_mapping = MyUserDict(original_dict)
 
             _custom_protobuf_getattribute(mutable_mapping, "data")
+
+    def test_address_server_data(self):
+        with override_env({"DD_IAST_ENABLED": "True"}):
+            with override_global_config(dict(_asm_enabled=True)):
+                with override_config("grpc", dict(service_name="myclientsvc")):
+                    with override_config("grpc_server", dict(service_name="myserversvc")):
+                        with mock.patch("ddtrace.appsec._asm_request_context.set_waf_address") as mock_set_waf_addr:
```

Comment:
nit: Not sure if this is much cleaner, but would prevent the cascading indents by a bit?:

```suggestion
        with override_env({"DD_IAST_ENABLED": "True"}), \
             override_global_config(dict(_asm_enabled=True)), \
             override_config("grpc", dict(service_name="myclientsvc")), \
             override_config("grpc_server", dict(service_name="myserversvc")):
            with mock.patch("ddtrace.appsec._asm_request_context.set_waf_address") as mock_set_waf_addr:
```

Reply:
Probably cleaner, yes.